BGL/include/CGAL/boost/graph/copy_face_graph.h

```diff
@@ -223,62 +268,165 @@ void copy_face_graph(const SourceMesh& sm, TargetMesh& tm,
+  
+  \cgalNamedParamsBegin
+    \cgalParamBegin{vertex_point_map}
+      the property map with the points associated to the vertices of `sm` (`tm`).
+      If this parameter is omitted, an internal property map for
+      `CGAL::vertex_point_t` should be available in `SourceMesh` (`TargetMesh`)
+    \cgalParamEnd
+  \cgalNamedParamsEnd
 
   The points from `sm` to `tm` are converted using
   `CGAL::Cartesian_converter<SourceKernel, TargetKernel>`.
   `SourceKernel` and `TargetKernel` are deduced using `CGAL::Kernel_traits`
-  from the value types of `Src_vpm` and `Tgt_vpm`.
+  from the value types of the vertex_point_maps.
 
   Other properties are not copied.
 */
-#if defined(DOXYGEN_RUNNING) // Use template default arguments
 template <typename SourceMesh, typename TargetMesh,
-          typename V2V = Emptyset_iterator,
-          typename H2H = Emptyset_iterator,
-          typename F2F = Emptyset_iterator,
-          typename Src_vpm = typename boost::property_map<SourceMesh, vertex_point_t>::const_type,
-          typename Tgt_vpm = typename boost::property_map<TargetMesh, vertex_point_t>::type>
+          #ifndef DOXYGEN_RUNNING
+          typename T1, typename Tag1, typename Base1, 
+          typename T2, typename Tag2, typename Base2
+          #else
+          typename NamedParameters1, typename NamedParameters2
+          #endif
+          >
 void copy_face_graph(const SourceMesh& sm, TargetMesh& tm,
-                     V2V v2v = V2V(), H2H h2h = H2H(), F2F f2f = F2F(),
-                     Src_vpm sm_vpm = get(vertex_point, sm),
-                     Tgt_vpm tm_vpm = get(vertex_point, tm) )
-#else // use the overloads
+                     #ifndef DOXYGEN_RUNNING
+                     const CGAL::cgal_bgl_named_params<T1,Tag1,Base1>& np1,
+                     const CGAL::cgal_bgl_named_params<T2,Tag2,Base2>& np2
+                     #else
+                     const NamedParameters1& np1,
+                     const NamedParameters2& np2
+                     #endif
+                     )
+{
+  using boost::choose_param;
+  if (boost::is_default_param(get_param(np1, internal_np::vertex_to_vertex_output_iterator)))
+  {
+    if (boost::is_default_param(get_param(np1, internal_np::vertex_to_vertex_map))){
+      copy_face_graph(sm, tm, CGAL::parameters::
+                      vertex_to_vertex_output_iterator(
+                        boost::make_function_output_iterator(internal::make_functor(get_param(np1, internal_np::vertex_to_vertex_map))))
+                      .halfedge_to_halfedge_output_iterator(get_param(np1, internal_np::halfedge_to_halfedge_output_iterator))
+                      .face_to_face_output_iterator(get_param(np1, internal_np::face_to_face_output_iterator))
+                      .halfedge_to_halfedge_map(get_param(np1, internal_np::halfedge_to_halfedge_map))
+                      .face_to_face_map(get_param(np1, internal_np::face_to_face_map))
```

Comment:
why not simply `(sm, tm, np1.vertex_to_vertex_output_iterator(                   boost::make_function_output_iterator(internal::make_functor(get_param(np1, internal_np::vertex_to_vertex_map)))), np2)`